src/os/linux/linux_sigar.c

```diff
@@ -246,6 +246,63 @@ char *sigar_os_error_string(sigar_t *sigar, int err)
     return NULL;
 }
 
+char * sigar_get_machine_id(void) {
+  struct utsname utsbuf;
+  struct dirent *data;
+  static char machine_id[256] = "";
+  static char tbuf[64] = "";
+  FILE *fp;
+
+  /* only generate once */
+
+  if (strlen(machine_id)) {
+    return machine_id;
+  }
+
+  /* get host name */
+
+  if (uname(&utsbuf) == 0) {
```

Comment:
I might suggest adding a void sigar_get_machine_id_common(char machine_id[256]) that calls sigar_net_info_get or sigar_sys_info_get and fills it in with host_name / domain_name.

That will work on all of the platforms out of the box, and allow your default implementation stubs for the other OSes to work a bit more.

Come to think of it, you could even just add a machine_id field to sigar_sys_info_t and move the implementation specifics there. What do you think?